Properly escape pipe character on item lore

imDaniX opened the pull request:
Using negative lookbehind instead of negated characters set.
Fixes #2962, #2984

Yeah, not the best solution, but it fixes the problem.

imDaniX committed:
Properly escape pipe character on item lore

Using negative lookbehind instead of negated characters set.
Fixes #2962, #2984

imDaniX committed:
Whoops

imDaniX commented:
Seems like it is the same problem #2953

pop4959 commented:
Thanks for the PR. It seems this bug was introduced in #2831.

The negative look-behind seems to be an appropriate enough fix for the issue. Have you done any testing on your changes to ensure they are working as expected? If not, I would at least try [LazyGon's test case](https://github.com/EssentialsX/Essentials/issues/2962#issuecomment-577163834) from the main issue thread.

imDaniX commented:
I tested it, but without a plugin itself.
```java
String lore = "|testing123.|test123.|test3.";
// old
for(String line : lore.split("[^\\\\](\\|)"))
    System.out.println(">" + line.replace('_', ' ') + "<");
System.out.println();
// new
for(String line : lore.split("(?<!\\\\)\\|"))
    System.out.println(">" + line.replace('_', ' ').replace("\\|", "|") + "<");
```
prints
```
>|testing123<
>test123<
>test3.<

><
>testing123.<
>test123.<
>test3.<
```
